### ISSUE TITLE
Image uploading failing (intermittently) with 500 error prior to hitting ImageUploadsController?

### ISSUE THREAD

**Issue by mikew1** [writingink/wink#148]
On the image uploader, upload only succeeds on "Crop Image", not on "Use Original".   
Also, .png works, .jpg doesn't at all (of the two test images I'm using).

So, I've got images uploading successfully, but not reliably.

I added a Log::info in the ImageUploadsController, which reveals on case of upload fails browser console reports 500 internal server error prior to ImageUploadsController being hit at all. Wondering what could that be... 

Would love to get this working & get comfortable with the source.  It's very elegantly written - the laravel parts which I've read.   :)

**Comment by devcircus** [comment]
Just a guess but it sounds like an upload size issue from php. That would throw a 500 before you even hit the controller. Maybe google "upload_max_filesize" and try increasing it in your php.ini.

**Comment by mikew1** [comment]
Mine's set at 2M, and the file is 50kb. This is on nginx with valet, I could switch over to an apache vhost if I thought that would help, as I have both setups running pointing at the same php-fpm via a unix socket.  

I've just been looking at `CropperModal.vue` in `resources/js/components` in the wink source. There are two methods there, `uploadOriginalImage()` and `uploadSelectedImage()`. These do very similar things but seem implemented differently.  Do you think that code needs a refactor?  I've been adding some console.logs in it to try to get my bearings.

**Comment by devcircus** [comment]
Interesting. I'll install a fresh app, check out the code  and see if I can reproduce.

**Comment by mikew1** [comment]
This is the error I'm still getting.  Problem is I don't know how to diagnose this.

![screen shot 2019-02-06 at 14 58 38](https://user-images.githubusercontent.com/8282106/52350141-ce990b80-2a1f-11e9-8105-55485847cb47.png)

**Comment by mikew1** [comment]
A bit more info.  I was getting the Alert.vue component triggered by alertError (relevant section of app.js shown below), but just got the svg "X", with no error message, so investigated as below.  So looks like httpError but error msg is undefined.  Wondering how is wink is doing api auth?   
![screen shot 2019-02-06 at 15 15 23](https://user-images.githubusercontent.com/8282106/52351385-8deec180-2a22-11e9-93a8-4ca8dd01185f.png)

![screen shot 2019-02-06 at 15 15 29](https://user-images.githubusercontent.com/8282106/52351353-80d1d280-2a22-11e9-97cf-ba44e4a15653.png)

**Comment by mikew1** [comment]
And in vue dev tools...  please bear with me while I learn to debug here :)
![screen shot 2019-02-06 at 16 57 25](https://user-images.githubusercontent.com/8282106/52358753-7b7b8480-2a30-11e9-8387-9844a477fe5d.png)

**Comment by mikew1** [comment]
Apologies everyone, it was nginx permissions problem so it couldn't save the image. I finally diagnosed this issue by actually looking at the nginx logs. You find this by `brew info nginx` which will lead you to where they are.   On finding that I chose to switch to apache which I'm more familar with. 

If anyone has problems in future I recommend, in the directory of the app into which you have installed wink:
`cd public`
`ls -la`
if there is no `storage` link entry there you must obviously do `php artisan storage link`, as stated in CONTRIBUTING.md  Also, if that link exists, check it's to the right place. It's an absolute link and if you've moved directory locations it will be pointing to the wrong place.
Finally, permissions. This depends on your setup. It's the standard laravel stuff.  I seem to have boobied there by assuming valet would just integrate to my setup without further thought.

Now properly rolling with wink!   Hurra!
